app/src/main/java/io/pslab/activity/CreateConfigActivity.java

```diff
@@ -0,0 +1,170 @@
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+                intervalUnit = "sec";
+            }
+        });
+
+        createConfigFileBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                interval = intervalEditText.getText().toString();
+                if (interval.length() == 0) {
+                    Toast.makeText(CreateConfigActivity.this, getResources().getString(R.string.no_interval_message), Toast.LENGTH_SHORT).show();
+                } else {
+                    SparseBooleanArray selectedParams = paramsListView.getCheckedItemPositions();
+                    ArrayList<String> selectedParamsList = new ArrayList<>();
+                    for (int i = 0; i < paramsListView.getCount(); i++) {
+                        if (selectedParams.get(i)) {
+                            selectedParamsList.add(instrumentParamsList.get(selectedItem)[i]);
+                        }
+                    }
+                    createConfigFile(selectedParamsList);
+                }
+            }
+        });
+    }
+
+    private void createArrayLists() {
+
+        instrumentParamsList.add(getResources().getStringArray(R.array.oscilloscope_params));
+        instrumentParamsList.add(getResources().getStringArray(R.array.multimeter_knob_states));
+        instrumentParamsList.add(getResources().getStringArray(R.array.logic_analyzer_params));
+        instrumentParamsList.add(getResources().getStringArray(R.array.barometer_params));
+        instrumentParamsList.add(getResources().getStringArray(R.array.luxmeter_params));
+        instrumentParamsList.add(getResources().getStringArray(R.array.accelerometer_params));
+
+        instrumentsList.add(getResources().getString(R.string.oscilloscope));
+        instrumentsList.add(getResources().getString(R.string.multimeter));
+        instrumentsList.add(getResources().getString(R.string.logical_analyzer));
+        instrumentsList.add(getResources().getString(R.string.baro_meter));
+        instrumentsList.add(getResources().getString(R.string.lux_meter));
+        instrumentsList.add(getResources().getString(R.string.accelerometer));
+    }
+
+    private void createCheckboxList() {
+        paramsListView.setAdapter(new ArrayAdapter<String>(CreateConfigActivity.this, android.R.layout.simple_list_item_multiple_choice, instrumentParamsList.get(selectedItem)));
+        paramsListView.setChoiceMode(ListView.CHOICE_MODE_MULTIPLE);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void createConfigFile(ArrayList<String> params) {
+        String instrumentName = instrumentsList.get(selectedItem);
+        String uniqueFileName = CSVLogger.FILE_NAME_FORMAT.format(new Date());
```

Comment:
Make the file name `pslab_config.txt`

Reply:
Shouldn't we add timestamp to filename? Because if the user creates multiple files they will be overridden in the mobile storage.

Reply:
Why multiple files?

Reply:
If I want one config file for multimeter just record RES and CAP and another for CH1 and CH2. I might want to use them at different time but ,at least I should be able to create them . That's what i think. Correct me if i have misunderstood something.

Reply:
How can I read a config file with a random name from hardware?

Reply:
I there any way you can check the first 2 parts of the file name from firmware and take the file as a config file. For example we can have file name as ``` pslab_config_<timestamp>.txt ``` . That way we can have multiple files in mobile storage as well. Just brainstorming, don't know if this is possible on hardware side.

Reply:
Don't make things complicated. Just do the requested change.

Reply:
Got it. :+1: